apps/src/applab/manageAssets.jsx

```diff
@@ -0,0 +1,138 @@
+    }.bind(this), function (xhr) {
+      this.setState({uploadStatus: 'Error loading asset list: ' +
+          getErrorMessage(xhr.status)});
+    }.bind(this));
+  },
+
+  fileUploadClicked: function () {
+    var uploader = document.querySelector('#uploader');
+    uploader.click();
+  },
+
+  upload: function () {
+    var file = document.querySelector('#uploader').files[0];
+    var uploadStatus = document.querySelector('#uploadStatus');
+
+    if (file.type && this.props.typeFilter) {
+      var type = file.type.split('/')[0];
+      if (type !== this.props.typeFilter) {
+        this.setState({uploadStatus: 'Only ' + this.props.typeFilter +
+          ' assets can be used here.'});
+        return;
+      }
+    }
+
+    // TODO: Use Dave's client api when it's finished.
+    AssetsApi.ajax('PUT', file.name, function (xhr) {
+      this.state.assets.push(JSON.parse(xhr.responseText));
+      this.setState({uploadStatus: 'File "' + file.name +
+          '" successfully uploaded!'});
+    }.bind(this), function (xhr) {
+      this.setState({uploadStatus: 'Error uploading file: ' +
+          getErrorMessage(xhr.status)});
+    }.bind(this), file);
+
+    this.setState({uploadStatus: 'Uploading...'});
+  },
+
+  deleteAssetRow: function (name) {
+    this.setState({
+      assets: this.state.assets.filter(function (asset) {
+        return asset.filename !== name;
+      })
+    });
+    this.setState({uploadStatus: 'File "' + name + '" successfully deleted!'});
+  },
+
+  render: function () {
+    var assetList;
+    if (this.state.assets === null) {
+      assetList = (
+        <div style={{margin: '1em 0', textAlign: 'center'}}>
+          <i className="fa fa-spinner fa-spin" style={{fontSize: '32px'}}></i>
+        </div>
+      )
+    } else if (this.state.assets.length === 0) {
+      assetList = (
+        <div style={{margin: '1em 0'}}>
+          Your assets will appear here. Click "Upload File" to add a new asset
+          for this project.
+        </div>
+      );
+    } else {
+      assetList = (
+        <div style={{maxHeight: '330px', overflow: 'scroll', margin: '1em 0'}}>
+          <table style={{width: '100%'}}>
+            <tbody>
+              {this.state.assets.map(function (asset) {
+                var choose = this.props.assetChosen &&
+                    this.props.assetChosen.bind(this, AssetsApi.basePath + '/' +
+                    asset.filename);
+
+                return <AssetRow key={asset.filename} name={asset.filename}
+                    type={asset.category} size={asset.size} choose={choose}
+                    delete={this.deleteAssetRow.bind(this, asset.filename)}/>;
+              }.bind(this))}
+            </tbody>
+          </table>
+        </div>
+      );
+    }
+
+    var title = this.props.assetChosen
+        ? <p className="dialog-title">Choose Assets</p>
+        : <p className="dialog-title">Manage Asset</p>;
+    var accept = (this.props.typeFilter || '*') + '/*';
```

Comment:
We should clarify this in the Style Guide.  Right now it says "[Binary and Ternary line continuations should have the operator at the beginning of the line](https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#js-operator-continuations)".  Should `+` be on the continued line (like `.`) or on the previous line  (like `?` and `:`)?

```javascript
var a = myObject.doSomething()
    .doSomethingElse();

var b = "really long string"
    + "another long string"

var c = "really long string" +
    "another long string"

var d = condition ?
    option1 :
    option2
```